nifi-bootstrap/src/test/java/org/apache/nifi/bootstrap/process/RuntimeValidatorCheckerTest.java

```diff
@@ -0,0 +1,207 @@
+        checker = new RuntimeValidatorChecker(configurationClasses);
+
+        final List<RuntimeValidatorResult> results = checker.check();
+        assertEquals(5, results.size());
+        final List<RuntimeValidatorResult> failures = getFailures(results);
+        assertEquals(0, failures.size());
+    }
+
+    @Test
+    public void testAllFailuresEmptyFiles() throws IOException {
+        final List<RuntimeValidator> configurationClasses = new ArrayList<>();
+        final File emptyFile = getTempFile("empty_file", "");
+        configurationClasses.add(new AvailablePorts(emptyFile));
+        configurationClasses.add(new FileHandles(emptyFile));
+        configurationClasses.add(new ForkedProcesses(emptyFile));
+        configurationClasses.add(new Swappiness(emptyFile));
+        configurationClasses.add(new TimedWaitDuration(emptyFile));
+        checker = new RuntimeValidatorChecker(configurationClasses);
+
+        final List<RuntimeValidatorResult> results = checker.check();
+        assertEquals(5, results.size());
+        final List<RuntimeValidatorResult> failures = getFailures(results);
+        assertEquals(5, failures.size());
+        for (final RuntimeValidatorResult failure : failures) {
+            assertTrue(failure.getExplanation().contains("parse"));
+        }
+    }
+
+    @Test
+    public void testAllFailuresUnparsable() throws IOException {
+        final List<RuntimeValidator> configurationClasses = new ArrayList<>();
+        final File unparsableFile = getTempFile("unparsable", "abcdefghijklmnopqrstuvwxyz");
+        configurationClasses.add(new AvailablePorts(unparsableFile));
+        configurationClasses.add(new FileHandles(unparsableFile));
+        configurationClasses.add(new ForkedProcesses(unparsableFile));
+        configurationClasses.add(new Swappiness(unparsableFile));
+        configurationClasses.add(new TimedWaitDuration(unparsableFile));
+        checker = new RuntimeValidatorChecker(configurationClasses);
+
+        final List<RuntimeValidatorResult> results = checker.check();
+        assertEquals(5, results.size());
+        final List<RuntimeValidatorResult> failures = getFailures(results);
+        assertEquals(5, failures.size());
+        for (final RuntimeValidatorResult failure : failures) {
+            assertTrue(failure.getExplanation().contains("parse"));
+        }
+    }
+
+    @Test
+    public void testCannotFindFilesForConfiguration() {
+        final List<RuntimeValidator> configurationClasses = new ArrayList<>();
+        final File missingFile = new File("missing_file");
+        configurationClasses.add(new AvailablePorts(missingFile));
+        configurationClasses.add(new FileHandles(missingFile));
+        configurationClasses.add(new ForkedProcesses(missingFile));
+        configurationClasses.add(new Swappiness(missingFile));
+        configurationClasses.add(new TimedWaitDuration(missingFile));
```

Comment:
It looks like the configuration of configuration classes could be pulled out to a private method and reused, passing the test file reference and returning the List.